tests/Game.class.test.js

```diff
@@ -0,0 +1,916 @@
+      const game2048 = new Game();
+
+      const cellPositions = new Map();
+
+      for (let i = 0; i < 100; i++) {
+        game2048.start();
+
+        const state = game2048.getState();
+
+        state.forEach((row, rowIndex) => {
+          row.forEach((cell, cellIndex) => {
+            if (cell !== 0) {
+              const key = `${rowIndex}${cellIndex}`;
+
+              if (cellPositions.has(key)) {
+                cellPositions.set(key, cellPositions.get(key) + 1);
+              } else {
+                cellPositions.set(key, 1);
+              }
+            }
+          });
+        });
+      }
+
+      expect(cellPositions.size >= 8).toBe(true);
+      expect(cellPositions.size <= 16).toBe(true);
+    });
+  });
+
+  describe('board movement methods', () => {
+    describe('moveLeft', () => {
+      it('should not move the board if the game is not started', () => {
+        const game2048 = new Game([
+          [8, 0, 0, 0],
+          [0, 16, 0, 0],
+          [0, 0, 32, 0],
+          [0, 0, 0, 64],
+        ]);
+
+        game2048.moveLeft();
+
+        const state = game2048.getState();
+
+        expect(state).toEqual([
+          [8, 0, 0, 0],
+          [0, 16, 0, 0],
+          [0, 0, 32, 0],
+          [0, 0, 0, 64],
+        ]);
+      });
+
+      it('should move all non-zero cells to the left', () => {
+        const initialState = [
+          [8, 0, 0, 0],
+          [0, 16, 0, 0],
+          [0, 0, 32, 0],
+          [0, 0, 0, 64],
+        ];
+
+        const game2048 = new Game(cloneState(initialState));
+
+        game2048.start();
+        game2048.moveLeft();
+
+        const state = game2048.getState();
+
+        state.forEach((row, rowIndex) => {
+          const cursor = initialState[rowIndex].find((cell) => cell !== 0);
+
+          expect(checkIsCorrectlyAligned(row, 'start', cursor)).toBe(true);
+        });
+      });
+
+      it('should add a random cell with 2 or 4 at an empty position after the move', () => {
+        const initialState = [
+          [8, 0, 16, 0],
+          [0, 16, 0, 32],
+          [8, 0, 32, 0],
+          [0, 16, 0, 64],
+        ];
+
+        const game2048 = new Game(cloneState(initialState));
+
+        game2048.start();
+
+        const stateAfterStart = [...game2048.getState()];
+
+        game2048.moveLeft();
+
+        const stateAfterMove = game2048.getState();
+
+        const addedCells = stateAfterMove
+          .reduce((acc, row, rowIndex) => {
+            const countBeforeMove = stateAfterStart[rowIndex].filter((cell) => cell !== 0).length;
+
+            if (countBeforeMove === 4) {
+              return acc;
+            }
+
+            const newCells = row.slice(countBeforeMove).filter((cell) => cell !== 0);
+
+            return [
+              ...acc,
+              ...newCells,
+            ];
+          }, []);
+
+        expect(addedCells.length).toBe(1);
+        expect([2, 4].includes(addedCells[0])).toBe(true);
```

Comment:
You don't need all this, because you definitelly know that a new 2 of for will appear in the 3rd column 

So you can just check that 3rd column has 1 non empty cell and is it 2 or 4

Reply:
Why? I have 2 free slots in every row. After `start()` we will get 2 random cells somewhere on the board. I don't know where. After `moveLeft()` we will have such a state, for example:
```json
[
  [8, 2, 16, 0],
  [16, 32, 0, 0],
  [8,32, 0, 0],
  [4, 16, 64, 0],
]
```

So, the game will add a new cell somewhere on the free space, 3rd or 4th column.

My test logic finds that cell based on the two state changes.